### PR TITLE
Fixed incorrect datetime in block, ref #1525

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

#1525 was added to v20 only, because it was a BC breaking change. This PR solves that issue w/o changing existing methods signature. (code from 1525)

Not sure about this change ... changing ...

```
if (!($date instanceof Zend_Date) && $date && !strtotime($date)) {
    return '';
}
if (is_null($date)) {
    $date = Mage::app()->getLocale()->date(Mage::getSingleton('core/date')->gmtTimestamp(), null, null);
```

### Related Pull Requests

1. See OpenMage/magento-lts#1525

### Fixed Issues (if relevant)

1. Closes OpenMage/magento-lts#2850

### Manual testing scenarios (*)

1. see related PR
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by sreichel** [comment]
possibly related to #2739

**Comment by fballiano** [comment]
@kiatng what do you think about this?

**Comment by kiatng** [comment]
This is a better fix of the original issue. 

Is this going to be in v20? If it is and if there are custom code in the wild that uses `$useTimezone` in `formatDate()`, it'll produce unexpected result. But BC in v20 is tolerable. Is it worthwhile to add a note in README for BC in v20?

**Comment by kiatng** [comment]
Test result comparing original v19 `formatDate($date)` and `formatTimezoneDate($date)`:

1. `$date = '2022-12-28 13:00:00';`, both results are "28/12/2022 21:00", correctly converted to local time.
2. `$date = null;`, both results are "28/12/2022 03:04" (GMT time)
3. `$date = '2022-12-28 13:00:00'; $date = new Zend_Date($date);`, both results are "28/12/2022 13:00", correct because timezone in `Zend_Date` was not set.
4. `$date = '2022-12-28 13:00:00'; $date = strtotime($date);`, `formatDate()` -> "", `formatTimezoneDate()` -> "28/12/2022 21:00". I think result of `formatTimezoneDate()` is preferable.

File: app/code/core/Mage/Core/Helper/Data.php [hunk 1]
```diff
@@ -153,31 +153,50 @@ public function formatPrice($price, $includeContainer = true)
     /**
      * Format date using current locale options and time zone.
      *
-     * @param   string|Zend_Date|null $date
+     * @param   string|Zend_Date|null $date   If empty, return current datetime.
```
Comment (kiatng) [inline app/code/core/Mage/Core/Helper/Data.php:156]:
```suggestion
     * @param   string|Zend_Date|int|null $date If empty, return current local datetime.
```

**Comment by sreichel** [comment]
@kiatng thanks. Would be nice to had a unit-test here.

File: app/code/core/Mage/Core/Helper/Data.php [hunk 2]
```diff
@@ -153,31 +153,50 @@ public function formatPrice($price, $includeContainer = true)
...
      *
-     * @param   string|Zend_Date|null $date
+     * @param   string|Zend_Date|int|null $date If empty, return current local datetime.
      * @param   string              $format   See Mage_Core_Model_Locale::FORMAT_TYPE_* constants
      * @param   bool                $showTime Whether to include time
      * @return  string
      */
     public function formatDate($date = null, $format = Mage_Core_Model_Locale::FORMAT_TYPE_SHORT, $showTime = false)
     {
+        return $this->formatTimezoneDate($date, $format, $showTime);
```
Comment (kiatng) [inline app/code/core/Mage/Core/Helper/Data.php:163]: For v20 compatibility, maybe something like this:
```suggestion
        return func_num_args() === 4
             ? $this->formatTimezoneDate($date, $format, $showTime, func_get_args()[3]);
             : $this->formatTimezoneDate($date, $format, $showTime);
```

Comment (sreichel) [reply app/code/core/Mage/Core/Helper/Data.php:163]: Would say no. v20 change was added four month ago in 08/2022 .... if someone wrote code that uses 4th param, he/she should be able to change that line. _imho_

**Review by kiatng — APPROVED** [review]
Should not have serious effect in v20.

**Comment by fballiano** [comment]
so the idea is to port it to v20 right? otherwise it would be actually worse to maintain

**Comment by sreichel** [comment]
Add @kiatng fix to v19 and port it to v20 was my plan ... :grey_question:

**Comment by ADDISON74** [comment]
Closing this PR was wrong decission. It was a better fix for that issue.

**Comment by sreichel** [comment]
@ADDISON74 there is a lollow-up PR. (#2940)

**Pull request closed without merging**